packages/sources/elwood/src/adapter.ts

```diff
@@ -0,0 +1,110 @@
+import { AxiosRequestConfig, Builder, Logger, Validator } from '@chainlink/ea-bootstrap'
+import {
+  AdapterRequest,
+  APIEndpoint,
+  Config,
+  ExecuteWithConfig,
+  ExecuteFactory,
+  MakeWSHandler,
+  Requester,
+} from '@chainlink/ea-bootstrap'
+import { makeConfig } from './config'
+import * as endpoints from './endpoint'
+import { PriceResponse, ResponseMessage, SubscribeRequest } from './types'
+
+export const execute: ExecuteWithConfig<Config, endpoints.TInputParameters> = async (
+  request,
+  context,
+  config,
+) => {
+  return Builder.buildSelector<Config, endpoints.TInputParameters>(
+    request,
+    context,
+    config,
+    endpoints,
+  )
+}
+
+export const endpointSelector = (
+  request: AdapterRequest,
+): APIEndpoint<Config, endpoints.TInputParameters> =>
+  Builder.selectEndpoint<Config, endpoints.TInputParameters>(request, makeConfig(), endpoints)
+
+export const makeExecute: ExecuteFactory<Config, endpoints.TInputParameters> = (config) => {
+  return async (request, context) => execute(request, context, config || makeConfig())
+}
+
+const getPair = (input: AdapterRequest) => {
+  const validator = new Validator(input, endpoints.price.inputParameters, {})
+  const base = validator.validated.data.base.toUpperCase()
+  const quote = validator.validated.data.quote.toUpperCase()
+  return `${base}-${quote}`
+}
+
+const getSubscribeRequest = (
+  pairSymbol: string,
+  action: 'subscribe' | 'unsubscribe',
+): SubscribeRequest => ({
+  action: action,
+  stream: 'index',
+  symbol: pairSymbol,
+  index_freq: 1_000, //ms
+})
+
+export const makeWSHandler =
+  (
+    config?: Config,
+  ): MakeWSHandler<
+    ResponseMessage | any // TODO: full WS message types
+  > =>
+  () => {
+    const wsConfig = config || makeConfig()
+
+    const handleSubscription =
+      (action: 'subscribe' | 'unsubscribe') =>
+      (input: AdapterRequest): SubscribeRequest | undefined => {
+        try {
+          const subscriptionMsg = getSubscribeRequest(getPair(input), action)
+
+          const data: AxiosRequestConfig = {
+            url: wsConfig.api?.baseURL,
+            headers: {
+              ...wsConfig.api?.headers,
+              apiKey: wsConfig.apiKey ?? '',
+            },
+            method: 'post',
+            data: subscriptionMsg,
+          }
+
+          Requester.request(data)
```

Comment:
Can you add some clarification as to what's happening here? It seems this is a WS handler, but we're also making a POST request without catching the response data?

Reply:
The Elwood API only allows using POST requests to subscribe to WS feeds, so this was my attempt to send the subscription request and return `subscriptionMsg` without needing to mess with the framework. The `Requester.request` call is async, so waiting for the return value would require `awaiting` in every instance of `wsHandler.subscribe` and `wsHandler.unsubscribe`. I'm aware it's not ideal, so we can investigate a longer-term solution if desired.

Reply:
Could you add this explanation to the code, so we don't lose this knowledge? :sweat_smile:

Reply:
Yeah for sure

Reply:
Done